src/EditorFeatures/Core/Implementation/TodoComment/AbstractTodoCommentIncrementalAnalyzer.cs

```diff
@@ -91,9 +87,31 @@ public Task DocumentResetAsync(Document document, CancellationToken cancellation
             }
         }
 
+        private async Task<IList<TodoComment>> GetTodoCommentsAsync(Document document, ImmutableArray<TodoCommentDescriptor> tokens, CancellationToken cancellationToken)
+        {
+            var client = await _workspace.GetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
+            if (client != null && !document.IsOpen())
+            {
+                // run todo scanner on remote host
+                return await client.RunCodeAnalysisServiceOnRemoteHostAsync<IList<TodoComment>>(
+                    document.Project.Solution, nameof(IRemoteTodoCommentService.GetTodoCommentsAsync),
+                    new object[] { document.Id, tokens }, cancellationToken).ConfigureAwait(false);
+            }
+
+            // No remote host support, use inproc service
+            var service = document.GetLanguageService<ITodoCommentService>();
```

Comment:
I like the pattern we have with other services where we name the methods like "RunXXXInCurrentProcess" or RunXXXInRemoteProcess.  Then, in the IRemoteXXX service, you actually call RunXXXInCurrentProcess.  That makes as much code overlap as possible.